src/library/Pagination/PageJumper.js

```diff
@@ -0,0 +1,70 @@
+/* @flow */
+import React, { PureComponent } from 'react';
+import { createStyledComponent } from '../styles';
+import { FlexItem } from '../Flex';
+import { FormField } from '../Form';
+import TextInput from '../TextInput';
+import type { Messages } from './Pagination';
+
+type Props = {
+  'aria-label'?: string,
+  currentPage: number,
+  inputRef: (node: ?HTMLInputElement) => void,
+  messages: Messages,
+  onPageChange: (currentPage: number) => void,
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  totalPages: number
+};
+
+const Root = createStyledComponent(FlexItem, ({ width }) => ({ width }));
+
+export default class PageJumper extends PureComponent<Props> {
+  render() {
+    const { messages, inputRef, size, ...restProps } = this.props;
+    const rootProps = {
+      ...restProps
+    };
+
+    if (!messages.pageJumper) {
+      throw new Error(
+        'Must provide messages.pageJumper when pageJumper={true}'
+      );
+    }
+
+    const { label, placeholder } = messages.pageJumper;
+    const inputProps = {
+      label,
+      hideLabel: true,
+      input: TextInput,
+      caption: label,
+      htmlSize: 5,
+      inputRef,
+      onChange: this.handlePageJump,
+      placeholder,
+      size
+    };
+
+    return (
+      <Root {...rootProps}>
+        <FormField {...inputProps} />
+      </Root>
+    );
+  }
+
+  handlePageJump = (event: SyntheticInputEvent<>) => {
+    const { onPageChange, totalPages } = this.props;
+    event.persist();
+    const updateCurrentPage = () => {
+      const currentPage = parseInt(event.target.value);
+      if (1 <= currentPage && currentPage <= totalPages) {
```

Comment:
Opinion:  I find this line strange to read.  `1 <= currentPage` vs.  `currentPage > 1`